src/sentences/mtw.rs

```diff
@@ -0,0 +1,111 @@
+use nom::{
+    character::complete::{char, one_of},
+    combinator::opt,
+    number::complete::float,
+    sequence::preceded,
+    IResult,
+};
+
+use crate::{parse::NmeaSentence, Error, SentenceType};
+
+/// MTW - Mean Temperature of Water
+///
+/// <https://gpsd.gitlab.io/gpsd/NMEA.html#_mtw_mean_temperature_of_water>
+///
+/// ```text
+///        1   2 3
+///        |   | |
+/// $--MTW,x.x,C*hh<CR><LF>
+/// ```
+/// 1:  Temperature, degrees
+/// 2:  Unit of Measurement, Celsius
+/// 3:  Mandatory NMEA checksum
+#[derive(Debug, PartialEq)]
+pub struct MtwData {
+    pub temperature: Option<f32>,
+    pub temperature_unit: Option<MtwUnit>,
```

Comment:
1. Let's just leave `temperature` and add a documentation block `///` that it is only "degrees Celsius"
2. use `double` (f64) instead. We've discussed this before and we should unify the api to always use `f64`.